addons/sale_crm/models/sale_order.py

```diff
@@ -12,4 +12,11 @@ class SaleOrder(models.Model):
         domain="[('type', '=', 'opportunity'), '|', ('company_id', '=', False), ('company_id', '=', company_id)]")
 
     def action_confirm(self):
-        return super(SaleOrder, self.with_context({k:v for k,v in self._context.items() if k != 'default_tag_ids'})).action_confirm()
+        res = super(SaleOrder, self.with_context({k: v for k, v in self._context.items() if k != 'default_tag_ids'})).action_confirm()
+        self.opportunity_id._update_revenues_from_so()
+        return res
+
+    def _message_post_after_hook(self, message, msg_vals):
+        if self.env.context.get('mark_so_as_sent'):
+            self.opportunity_id._update_revenues_from_so()
+        return super()._message_post_after_hook(message, msg_vals)
```

Comment:
Posting a message is not really a business method. Not sure what kind of hook you expect here but I don't think this should be used to trigger business update.